### DISCUSSION TITLE
如何开启opengl渲染,并获取fbo

### DISCUSSION THREAD

**Discussion by evehal** [CefView/QCefView#430] [Q&A] [answered]
如题,我想启用硬件加速渲染,并用fbo和其他纹理做混合

**Comment by tishion** [comment, marked as answer]
CEF120以上才支持

**Reply by evehal** [reply]
谢谢

**Reply by StackExplode** [reply]
> CEF120以上才支持

您好，我使用的是`cef_binary_127.3.5+g114ea2a+chromium-127.0.6533.120_windows64`的CEF，编译并引入QT后还是很卡，请问打开Opengl渲染需要什么其他的设置吗？

**Reply by tishion** [reply]
> > CEF120以上才支持
> 
> 您好，我使用的是`cef_binary_127.3.5+g114ea2a+chromium-127.0.6533.120_windows64`的CEF，编译并引入QT后还是很卡，请问打开Opengl渲染需要什么其他的设置吗？

你可能理解错了，我们所说的open GL是指CEF在OSR模式下传递给外部程序用于渲染的图像数。跟你的卡顿无关，OSR模式现在不支持硬件加速渲染，你可以使用NCW模式。

**Reply by StackExplode** [reply]
> > > CEF120以上才支持
> > 
> > 
> > 您好，我使用的是`cef_binary_127.3.5+g114ea2a+chromium-127.0.6533.120_windows64`的CEF，编译并引入QT后还是很卡，请问打开Opengl渲染需要什么其他的设置吗？
> 
> 你可能理解错了，我们所说的open GL是指CEF在OSR模式下传递给外部程序用于渲染的图像数。跟你的卡顿无关，OSR模式现在不支持硬件加速渲染，你可以使用NCW模式。

感谢您的回答，我是照着您官网的案例写的程序，案例中默认启用了OSR，我设定为不使用OSR后帧数上去了。但是根据Readme所述，NCW模式对Linux不支持，对跨平台有点难受呀。我自己测试了下OSR模式，我观察任务管理器中GPU确实在动的，照理是硬件加速了吧？但是用`https://webglsamples.org/aquarium/aquarium.html`这个网站测试帧数，似乎是被钳制在的30FPS，谷歌搜索也有人提到OSR 30 FPS cap。我以前做嵌入式的，对软件不太懂，能否请大佬给点提示、线索、资料我自己学习也行。
PS. 您的官网案例里一些函数比较老了貌似没有更新，例如js里现在叫`CallBridge`而不是`window.CefViewClient`了，大佬有空可以更新下。
PPS. 关于按键监听，资料太少了，我自己做了些尝试有一些问题，我新开一个discuss，还麻烦大佬有空解惑下。
